Update proposals-CG-WG.md

hlflanagan opened the pull request:
See https://github.com/w3c-fedid/Administration/issues/11

TallTed commented on proposals-CG-WG.md:
```diff
@@ -2,7 +2,7 @@
 
 Author: @samuelgoto
 Created: 08/11/2024
```
We're international. [ISO 8601](https://xkcd.com/1179/), please.
```suggestion
Created: 2024-08-11
```

TallTed replied:
Is this August 11 or November 8?

ISO 8601 is unambiguous, if used consistently.

TallTed commented on proposals-CG-WG.md:
```diff
@@ -2,7 +2,7 @@
 Author: @samuelgoto
 Created: 08/11/2024
-Last update: 09/11/2024
+Last update: 11/26/2024
```
As above.
```suggestion
Last update: 2024-11-26
```

TallTed commented on proposals-CG-WG.md:
```diff
@@ -29,70 +29,73 @@ The purpose of Stage 0 proposals is to allow anyone to raise and explore the Pro
 # Stage 2: Formalization
 
-The purpose of Stage 2 Proposals is to formally specify the best (and seemingly workable) alternative that was identified in the prior step: handle corner cases, integrate with other parts, reconcile with other proposals, and resolve the concerns identified at the entrance. The Proposal enters Stage 2 with a list of blocking issues to advance to the next stage and exits with all of the issues res ...
+The goal of Stage 2 is to refine the preferred solution into a detailed, cohesive proposal that addresses known issues and integrates feedback from stakeholders. The Proposal enters Stage 2 with a list of blocking issues to advance to the next stage and exits with all of the issues resolved.  This stage focuses on preparing the proposal for Working Group review and creating a complete, formal dra ...
```
```suggestion
The goal of Stage 2 is to refine the preferred solution into a detailed, cohesive proposal that addresses known issues and integrates feedback from stakeholders. The Proposal enters Stage 2 with a list of issues that block advancement to the next stage, and exits with all of the issues resolved.  This stage focuses on preparing the proposal for Working Group review and creating a complete, formal draft.
```

TallTed replied:
Existing language is not proper English construction. I suggested changes to make it proper English. Why is this marked resolved without application of those changes?

TallTed commented on proposals-CG-WG.md:
```diff
@@ -29,70 +29,73 @@ The purpose of Stage 0 proposals is to allow anyone to raise and explore the Pro
+> will change a lot as we learn, including potentially entirely replacing them with references to the W3C Process. 
 
-The purpose of Stage 3 Proposals is to increase implementation and deployment confidence in order to produce a [Candidate Recommendation](https://www.w3.org/policies/process/#RecsCR).
+The purpose of Stage 3 Proposals is to increase implementation and deployment confidence in order to produce a [Candidate Recommendation](https://www.w3.org/policies/process/#RecsCR). The spec should be considered finished, pending editorial nit review. However, since multiple implementations are expected in this stage, it is possible there will be further changes to normative content.
```
```suggestion
The purpose of Stage 3 Proposals is to increase implementation and deployment confidence in order to produce a [Candidate Recommendation](https://www.w3.org/policies/process/#RecsCR). The spec should be considered finished, pending editorial nit review. However, since multiple implementations are expected to be produced during this stage, it is possible there will be further changes to normative content based on feedback from those implementors.
```

TallTed commented on proposals-CG-WG.md:
```diff
@@ -29,70 +29,73 @@ The purpose of Stage 0 proposals is to allow anyone to raise and explore the Pro
 # Stage 4: Publication
 
- The purpose of Stage 4 Proposals is to produce a [W3C Recommendation](https://www.w3.org/policies/process/#RecsW3C).
+ The purpose of Stage 4 Proposals is to produce a [W3C Recommendation](https://www.w3.org/policies/process/#RecsW3C). At this stage, the spec is merged and has finished editor review. This editor review could be lengthy, especially if the feature is large and/or the contributor is new to W3C process, but it will usually be short.
```
```suggestion
 The purpose of Stage 4 Proposals is to produce a [W3C Recommendation](https://www.w3.org/policies/process/#RecsW3C). At this stage, the spec is merged and has finished editorial review. This editorial review could be lengthy, especially if the feature is large and/or the contributor is new to W3C process, but it will usually be short.
```

TallTed reviewed (changes requested):
Small, for clarity

timcappalli reviewed (approved):
lgtm

simoneonofri reviewed (approved):
LGTM

samuelgoto commented on proposals-CG-WG.md:
```diff
@@ -29,70 +29,73 @@ The purpose of Stage 0 proposals is to allow anyone to raise and explore the Pro
-    * [ ] Confidence of developer demand and fitness for purpose (e.g. a developer that needs this proposal)
+    * [ ] An [explainer](https://tag.w3.org/explainers/).
+    * [ ] Documentation of alternatives and trade-offs considered.
+    * [ ] Draft specification text (or detailed examples/code samples if needed for clarity).
```
This seems like a higher bar than is necessary. I'd suggest allowing proposals to enter Stage 2 without a draft specification text (explainers are necessary).

```suggestion
    * [ ] Optionally, a draft specification text (or detailed examples/code samples if needed for clarity).
```

samuelgoto commented on proposals-CG-WG.md:
```diff
@@ -29,70 +29,73 @@ The purpose of Stage 0 proposals is to allow anyone to raise and explore the Pro
+    * [ ] Documentation of alternatives and trade-offs considered.
+    * [ ] Draft specification text (or detailed examples/code samples if needed for clarity).
+    * [ ] Early implementation experience, including prototypes or trials.
+    * [ ] Evidence of stakeholder demand and use-case alignment.
```
Stakeholders seems unnecessarily underspecified as far as "demand" is concerned. Where else would "demand" come from other than web developers?

```suggestion
    * [ ] Evidence of stakeholder (e.g. web developers) demand and use-case alignment.
```

TallTed replied:
Minor tweaks to @samuelgoto's suggestion
```suggestion
    * [ ] Evidence of stakeholder (e.g., web developer) demand and use-case alignment.
```

samuelgoto commented on proposals-CG-WG.md:
```diff
@@ -29,70 +29,73 @@ The purpose of Stage 0 proposals is to allow anyone to raise and explore the Pro
-    * [ ] An [Editor's Draft](https://www.w3.org/policies/process/#editors-draft) can be used to get PRs merged between [Working Draft](https://www.w3.org/policies/process/#RecsWD) revisions.
+    * [ ] Working Group consensus to adopt the proposal as the basis for their work.
+    * [ ] A clear list of blocking issues to be addressed before advancing to Stage 3.
+    * [ ] A completed Working Draft for further iteration.
```
What does this mean? What does it mean that we ask the Working Group for a Working Draft?

That is, do we expect every different proposal to have separate working drafts?

npm1 replied:
As a spec editor, I would not enjoy having a separate 'working draft' for every feature. That said, maybe this can say 'pull request' at least for features that are intended to be part of an existing spec and not their own spec (which are probably the vast majority)?

hlflanagan replied:
If it's an update, then yes, a PR. If it's a whole new thing, then a doc draft.

samuelgoto commented on proposals-CG-WG.md:
```diff
@@ -29,70 +29,73 @@ The purpose of Stage 0 proposals is to allow anyone to raise and explore the Pro
+    * [ ] Working Group consensus to adopt the proposal as the basis for their work.
+    * [ ] A clear list of blocking issues to be addressed before advancing to Stage 3.
+    * [ ] A completed Working Draft for further iteration.
+    * [ ] Approval to transition the draft into a formal as the basis for their work as a [Working Draft](https://www.w3.org/policies/process/#RecsWD).
```
I don't follow this: do we expect that, when a proposal enters Stage 2, that it will have the proposal merged into the Working Draft?

What does this mean? What does it mean that we ask the Working Group for a Working Draft?

That is, do we expect every different proposal to have separate working drafts?

Or is it that what one gets is approval to spend time in the WG to review spec PRs?

```suggestion
    * [ ] Approval to dedicate the WG's time to Review spec PRs to merge the proposal into the [Working Draft](https://www.w3.org/policies/process/#RecsWD).
```

samuelgoto commented on proposals-CG-WG.md:
```diff
@@ -29,70 +29,73 @@ The purpose of Stage 0 proposals is to allow anyone to raise and explore the Pro
-    * [ ] A specific preferred proposal, e.g., code samples, examples, or a spec PRs (if the formalism is needed to understand the proposal)
-    * [ ] Browser and developer implementation experience (e.g., a prototype, dev trials, origin trials, etc) 
-    * [ ] Confidence of developer demand and fitness for purpose (e.g. a developer that needs this proposal)
+    * [ ] An [explainer](https://tag.w3.org/explainers/).
```
Can I suggest that we keep these editorial fixes outside of this PR so that we can focus on the normative changes? For example, I almost dismissed the change below because I assumed that it was just making editorial changes rather than normative.

The editorial changes are perfectly valid, I just think that they may be best done in isolation rather than together with this spec PR.

samuelgoto replied:
can you revert the non-editorial changes so that it is easier to review?

TallTed replied:
> The editorial changes are perfectly valid, I just think that they may be best done in isolation rather than together with this spec PR.

Often, editorial changes make a significant difference to the comprehensibility of a spec, which makes a difference in reviewing the technical changes. Certainly, editorial changes of the lines impacted by the PR should be allowed, because such in-PR changes are a substantially lighter lift for the reviewer/commenter than a distinct PR — and don't require that the reviewer remember, once the technical PR is merged, that they wanted to make an editorial change.

npm1 commented on proposals-CG-WG.md:
```diff
@@ -29,70 +29,73 @@ The purpose of Stage 0 proposals is to allow anyone to raise and explore the Pro
 # Stage 2: Formalization
 
-The purpose of Stage 2 Proposals is to formally specify the best (and seemingly workable) alternative that was identified in the prior step: handle corner cases, integrate with other parts, reconcile with other proposals, and resolve the concerns identified at the entrance. The Proposal enters Stage 2 with a list of blocking issues to advance to the next stage and exits with all of the issues res ...
+The goal of Stage 2 is to refine the preferred solution into a detailed, cohesive proposal that addresses known issues and integrates feedback from stakeholders. The Proposal enters Stage 2 with a list of blocking issues to advance to the next stage and exits with all of the issues resolved.  This stage focuses on preparing the proposal for Working Group review and creating a complete, formal dra ...
```
Complete formal draft of what? I guess the specification of the feature?

hlflanagan replied:
Yes, specification of the feature.

elf-pavlik replied:
W3C CGs use *Draft Report* and *Final Report* terminology. Could **draft report** make it more clear?
https://www.w3.org/community/about/process/#deliverables
![image](https://github.com/user-attachments/assets/4f358a4b-a222-483c-92ee-44667cabd28c)

npm1 commented on proposals-CG-WG.md:
```diff
@@ -29,70 +29,73 @@ The purpose of Stage 0 proposals is to allow anyone to raise and explore the Pro
   * What's asked of the **Working Group**? 
-    * [ ] Working Group consensus that the [Working Draft](https://www.w3.org/policies/process/#RecsWD) sufficiently resolves all of the issues raised at [Stage 2](#stage-2)
-    * [ ] Working Group consensus to publish the [Working Draft](https://www.w3.org/policies/process/#RecsWD) as the Working Group's [Candidate Recommendation](https://www.w3.org/policies/process/#RecsCR)
+    * [ ] Working Group consensus that the [Working Draft](https://www.w3.org/policies/process/#RecsWD) sufficiently resolves all of the issues raised at [Stage 2](#stage-2).
```
Since we are modifying this, maybe we should address the issue about not all features within a spec reaching Stage 3 at the same time. Perhaps the CR mentioned below can be a frozen branch from the WD, possibly with features in lesser stages removed?

hlflanagan replied:
I don't think that's necessary, though if you feel strongly about it, we'll have to come up with text for both stage 2 and 3, as neither require all features reach the stage at the same time.

TallTed commented on proposals-CG-WG.md:
```diff
@@ -29,70 +29,73 @@ The purpose of Stage 0 proposals is to allow anyone to raise and explore the Pro
   * Examples  
-    * Decorators Repo: [https://github.com/tc39/proposal-decorators](https://github.com/tc39/proposal-decorators)   
-    * Temporal Repo: [https://github.com/tc39/proposal-temporal](https://github.com/tc39/proposal-temporal)   
+     * Decorators Repo: [https://github.com/tc39/proposal-decorators](https://github.com/tc39/proposal-decorators)   
+     * Temporal Repo: [https://github.com/tc39/proposal-temporal](https://github.com/tc39/proposal-temporal)   
```
Why the extra space for this third-tier indent, here and below? Note that prior levels (and the "original" being changed here) are using 2 spaces per tier, a la --
```
* top
  * second
    * third
```

wseltzer commented on proposals-CG-WG.md:
```diff
@@ -13,7 +13,7 @@ This is a proposal to break proposals into 5 stages of maturity, with clear guid
 * [Stage 4](#stage4): **Publication** of a Proposed Recommendation
 
-These stages support the W3C process. If there is ever any question between W3C process requirements and how the groups will progress their work, the W3C process has precedent.
+These stages support the W3C process and align with the [TC39 process](https://tc39.es/process-document/) and the [WHATWG process](https://whatwg.org/stages). If there is ever any question between W3C process requirements and how the groups will progress their work, the W3C process has precedent.
```
```suggestion
These stages support the W3C process and align with the [TC39 process](https://tc39.es/process-document/) and the [WHATWG process](https://whatwg.org/stages). If there is ever any question between W3C process requirements and how the groups will progress their work, the W3C process has precedence.
```

samuelgoto replied:
Can you revert this change too as it is covered in this other PR? https://github.com/w3c-fedid/Administration/pull/13/files#diff-413347616d7d3063a428e05c5c47d90aa1301ba3dade010736161ffb20774cedR16

hlflanagan committed:
Revert "Update proposals-CG-WG.md
"

This reverts commit e2019ccd966ec254923dc5b3f70afb6922d10653.

hlflanagan committed:
Revert "Update proposals-CG-WG.md
"

This reverts commit 4c332650f4b8a0044f7c1a28e184a0814c922214.

hlflanagan committed:
Revert "Update proposals-CG-WG.md
"

This reverts commit 36b00ebdab72386d2b519ea1788a76a8442bff57.

hlflanagan committed:
Revert "Update proposals-CG-WG.md (date format)"

This reverts commit 64dc551dd39bfc3337f0120be31d1f48fd6d2ef4.

hlflanagan committed:
Revert "Update proposals-CG-WG.md (date format)"

This reverts commit dc2c13704423bb0531cb89e43f590447ea760f73.

hlflanagan committed:
reverting purely editorial changes

npm1 commented on proposals-CG-WG.md:
```diff
@@ -38,42 +38,45 @@ The purpose of Stage 1 proposals is to explore the Solution Space and pick a pre
-    * [ ] An [Editor's Draft](https://www.w3.org/policies/process/#editors-draft) can be used to get PRs merged between [Working Draft](https://www.w3.org/policies/process/#RecsWD) revisions.
+    * [ ] Working Group consensus to adopt the proposal as the basis for their work.
+    * [ ] A clear list of blocking issues to be addressed before advancing to Stage 3.
+    * [ ] A completed Working Draft for further iteration.
```
Link to `Working Draft` seemed useful, why was it removed?

hlflanagan replied:
Ooops.

npm1 commented on proposals-CG-WG.md:
```diff
@@ -38,42 +38,45 @@ The purpose of Stage 1 proposals is to explore the Solution Space and pick a pre
-    * [ ] An [Editor's Draft](https://www.w3.org/policies/process/#editors-draft) can be used to get PRs merged between [Working Draft](https://www.w3.org/policies/process/#RecsWD) revisions.
+    * [ ] Working Group consensus to adopt the proposal as the basis for their work.
+    * [ ] A clear list of blocking issues to be addressed before advancing to Stage 3.
+    * [ ] A completed Working Draft for further iteration.
```
This seems at odds with the above `Optionally, a draft specification text (or detailed examples/code samples if needed for clarity).`. A Working Draft is a draft specification with minimal adoption from the WG, no? So how is the draft spec optional if the WD is required?

hlflanagan replied:
You're right. Reverting that change.

npm1 commented on proposals-CG-WG.md:
```diff
@@ -38,42 +38,45 @@ The purpose of Stage 1 proposals is to explore the Solution Space and pick a pre
-    * [ ] Further implementation experience
+    * [ ] Proposal fully merged into the [Working Draft](https://www.w3.org/policies/process/#RecsWD).
+    * [ ] Web Platform Tests are available.
+    * [ ] At least two independent implementers and no unresolved objections (where resolution may include an agreement not to address the issue).
```
This requirement seems a bit off compared with the previous text `since multiple implementations are expected to be produced during this stage`. So are multiple implementors needed to reach stage 3 or could they possibly occur during stage 3? Note that in WHATWG it is two implementers supporting which is different from having actually implemented. https://whatwg.org/stages#stage3

hlflanagan replied:
Agreed. Stage 3 is ready for implementation; so rephrasing to supporting makes sense.

npm1 replied:
Your agreement does not seem reflected in the text

npm1 commented on proposals-CG-WG.md:
```diff
@@ -38,42 +38,45 @@ The purpose of Stage 1 proposals is to explore the Solution Space and pick a pre
-    * [ ] An [Editor's Draft](https://www.w3.org/policies/process/#editors-draft) can be used to get PRs merged between [Working Draft](https://www.w3.org/policies/process/#RecsWD) revisions.
+    * [ ] Working Group consensus to adopt the proposal as the basis for their work.
+    * [ ] A clear list of blocking issues to be addressed before advancing to Stage 3.
+    * [ ] A completed Working Draft for further iteration.
```
Doesnt this mean that the spec PR is not optional as suggested above? How can you have "A completed Working Draft for further iteration" without "a draft specification text"

npm1 replied:
Or maybe Im just reading this wrong and the first is about prerequisites while the second is about exit criteria? I think "What's needed from champions?" and "What's asked of the **Working Group**?" is vague and should be made clearer about whether this is about prerequisites or exit criteria

hlflanagan committed:
Merge branch 'main' into issue-11

hlflanagan commented:
OK, after braving various git merge conflicts between the editorial PR that closed and the other PR with more substantive wording changes, I think I've resolved just about everything. I will go ahead and apologize in advance for the changes in Stage 4, which are, unfortunately, editorial but seemed to be part of my merge conflicts. @npm1, would you mind taking a look at the remaining wording changes? I think they've largely addressed your feedback.

samuelgoto reviewed (approved):
LGTM++

TallTed commented on proposals-CG-WG.md:
```diff
@@ -29,28 +29,28 @@ The purpose of Stage 0 proposals is to allow anyone to raise and explore the Pro
 # Stage 2: Formalization
 
-The goal of Stage 2 is to refine the preferred solution into a detailed, cohesive proposal that addresses known issues and integrates feedback from stakeholders. The Proposal enters Stage 2 with a list of blocking issues to advance to the next stage and exits with all of the issues resolved.  This stage focuses on preparing the proposal for Working Group review and creating a complete, formal dra ...
+The goal of Stage 2 is to refine the preferred solution into a detailed, cohesive proposal that addresses known issues and integrates feedback from stakeholders. The Proposal enters Stage 2 with a list of blocking issues to advance to the next stage and exits with all of the issues resolved.  This stage focuses on preparing the proposal for Working Group review and creating a complete, formal [Dr ...
```
```suggestion
The goal of Stage 2 is to refine the preferred solution into a detailed, cohesive proposal that addresses known issues and integrates feedback from stakeholders. The Proposal enters Stage 2 with a list of issues blocking advance to the next stage and exits with all of the issues resolved.  This stage focuses on preparing the proposal for Working Group review and creating a complete, formal [Draft Report](https://www.w3.org/community/about/process/#deliverables).
```

TallTed commented on proposals-CG-WG.md:
```diff
@@ -94,7 +94,7 @@ The purpose of Stage 3 Proposals is to increase implementation and deployment co
      * Depth sensing Repo: [https://github.com/immersive-web/depth-sensing](https://github.com/immersive-web/depth-sensing)   
 * WebAssembly  
-  * Proposals Repo: [https://github.com/WebAssembly/proposals](https://github.com/WebAssembly/proposals)  
+   * Proposals Repo: [https://github.com/WebAssembly/proposals](https://github.com/WebAssembly/proposals)  
```
```suggestion
  * Proposals Repo: [https://github.com/WebAssembly/proposals](https://github.com/WebAssembly/proposals)  
```

npm1 reviewed (approved):
Ok let me give what I understand is the TLDR the changes are:
* stage 2 no longer requires a merged pull request, just existence of one
* stage 3 no longer requires multiple implementations, just multiple implementors approving of the feature

Is that right? If so, seems fine with me.
